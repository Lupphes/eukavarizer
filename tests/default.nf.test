nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "pipeline"

    test("eukavarizer: test profile") {

        when {
            params {
                outdir = "$outputDir"
                input = "${projectDir}/tests/samplesheet_minimal.csv"
                reference_genome = "${projectDir}/tests/GCF_000146045.2_R64_genomic.fna.gz"
                taxonomy_id = 4932
                sniffles_flag = true
                minimap2_flag = true
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                path("$outputDir/4932/results/4932.html"),
                path("$outputDir/4932/ref/4932_refseq_results.json"),
                path("$outputDir/pipeline_info/nf_core_eukavarizer_software_mqc_versions.yml")
            ).match()

            // Check for presence of expected output directories and files
            assert path("$outputDir/4932").exists()
            assert path("$outputDir/pipeline_info").exists()

            // Check for timestamped pipeline_info files (can't use wildcards in snapshots)
            def pipelineInfoFiles = path("$outputDir/pipeline_info").list()
            assert pipelineInfoFiles.any { it.fileName.toString().startsWith("execution_report_") && it.fileName.toString().endsWith(".html") }
            assert pipelineInfoFiles.any { it.fileName.toString().startsWith("execution_timeline_") && it.fileName.toString().endsWith(".html") }
            assert pipelineInfoFiles.any { it.fileName.toString().startsWith("pipeline_dag_") && it.fileName.toString().endsWith(".html") }
        }
    }

    test("eukavarizer: all callers enabled") {

        when {
            params {
                outdir = "$outputDir"
                input = "${projectDir}/tests/samplesheet_minimal.csv"
                reference_genome = "${projectDir}/tests/GCF_000146045.2_R64_genomic.fna.gz"
                taxonomy_id = 4932
                minimap2_flag = false
                bwamem2 = false
                delly_flag = true
                manta_flag = true
                gridss_flag = true
                tiddit_flag = true
                sniffles_flag = true
                cutesv_flag = true
                dysgu_flag = true
                svaba_flag = true
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                path("$outputDir/4932/results/4932.html"),
                path("$outputDir/4932/ref/4932_refseq_results.json"),
                path("$outputDir/pipeline_info/nf_core_eukavarizer_software_mqc_versions.yml")
            ).match()

            // Check for presence of expected output directories and files
            assert path("$outputDir/4932").exists()
            assert path("$outputDir/pipeline_info").exists()

            // Check for timestamped pipeline_info files (can't use wildcards in snapshots)
            def pipelineInfoFiles = path("$outputDir/pipeline_info").list()
            assert pipelineInfoFiles.any { it.fileName.toString().startsWith("execution_report_") && it.fileName.toString().endsWith(".html") }
            assert pipelineInfoFiles.any { it.fileName.toString().startsWith("execution_timeline_") && it.fileName.toString().endsWith(".html") }
            assert pipelineInfoFiles.any { it.fileName.toString().startsWith("pipeline_dag_") && it.fileName.toString().endsWith(".html") }
        }
    }
}
