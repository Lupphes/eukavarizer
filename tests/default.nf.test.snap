{
    "eukavarizer: test profile": {
        "content": [
            "4932.html:md5,f8f6759006dfb9737346537ac28d013b",
            "4932_refseq_results.json:md5,86de71555dc6dc975720072420375548",
            "nf_core_eukavarizer_software_mqc_versions.yml:md5,01db1fc45ce48e99372b12d5e8ff35a3"
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.10.0"
        },
        "timestamp": "2025-10-25T23:06:19.695490574"
    },
    "eukavarizer: all callers enabled": {
        "content": [
            "4932.html:md5,1201ee4ce24d068690951870a178465c",
            "4932_refseq_results.json:md5,86de71555dc6dc975720072420375548",
            "nf_core_eukavarizer_software_mqc_versions.yml:md5,01db1fc45ce48e99372b12d5e8ff35a3"
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.10.0"
        },
        "timestamp": "2025-10-25T23:11:45.461999259"
    }
}